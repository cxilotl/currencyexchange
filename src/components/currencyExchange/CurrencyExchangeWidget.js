import React from 'react';
import cssStyles from './CurrencyExchangeWidget.module.scss';
import ExchangeRateDropdown from '../exchangheRateDropdown/ExchangeRateDropdown';
import CurrencyPocket from '../currencyPocket/CurrencyPocket';

const CurrencyExchangeWidget = () => {
  const accounts = [
    {
      currency: 'gbp',
      symbol: '£',
      balance: 58.33
    },
    {
      currency: 'eur',
      symbol: '€',
      balance: 116.12
    },
    {
      currency: 'usd',
      symbol: '$',
      balance: 45.5
    }
  ];
  const onInput = (value) => {
    console.log(value);
  };
  const exchangeRates = [
    `£1 = €1.3444`,
    `€1 = £0.7438`
  ];
  return(
    <div className={ cssStyles.widget }>
      <div className={ cssStyles.dropdown }>
        <ExchangeRateDropdown rates={ exchangeRates } />
      </div>
      <section className={ cssStyles.top }>
        <CurrencyPocket
          currency={ accounts[0].currency }
          currencySymbol={ accounts[0].symbol }
          balance={ accounts[0].balance }
          onChangeAmount={ onInput }
        />
      </section>
      <section className={ cssStyles.bottom }>
        <CurrencyPocket
          currency={ accounts[1].currency }
          currencySymbol={ accounts[1].symbol }
          balance={ accounts[1].balance }
          onChangeAmount={ onInput }
        />
      </section>
    </div>
  );
};

export default CurrencyExchangeWidget;

