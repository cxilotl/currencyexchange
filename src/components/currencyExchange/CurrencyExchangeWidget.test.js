import React from 'react';
import renderer from 'react-test-renderer';
import CurrencyExchangeWidget from "./CurrencyExchangeWidget";

describe('CurrencyExchangeWidget', () => {

  it('Snapshot', () => {
    const component = renderer.create(
      <CurrencyExchangeWidget/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});