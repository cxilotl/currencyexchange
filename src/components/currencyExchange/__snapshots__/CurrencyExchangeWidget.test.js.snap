// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyExchangeWidget Snapshot 1`] = `
<div
  className="widget"
>
  <div
    className="dropdown"
  >
    <div
      className="wrapper"
    >
      <select
        className="select"
        data-testid="exchange-rate"
        id="exchange-rate"
      >
        <option
          className="option"
          value="£1 = €1.3444"
        >
          £1 = €1.3444
        </option>
        <option
          className="option"
          value="€1 = £0.7438"
        >
          €1 = £0.7438
        </option>
      </select>
    </div>
  </div>
  <section
    className="top"
  >
    <section
      className="layout"
    >
      <form
        className="form"
      >
        <label
          className="label"
          htmlFor="currentCurrency"
        >
          GBP
          <input
            className="input"
            data-testid="currency-input"
            id="currentCurrency"
            name="currentCurrency"
            onChange={[Function]}
            pattern="/^-?[0-9]+(\\\\.[0-9][0-9]?)?$/g"
            placeholder="0.00"
            type="number"
          />
        </label>
      </form>
      <div
        className="layout"
      >
        <p
          className="text"
        >
          You have 
          <span
            data-testid="account-balance"
          >
            £
            58.33
          </span>
        </p>
      </div>
    </section>
  </section>
  <section
    className="bottom"
  >
    <section
      className="layout"
    >
      <form
        className="form"
      >
        <label
          className="label"
          htmlFor="currentCurrency"
        >
          GBP
          <input
            className="input"
            data-testid="currency-input"
            id="currentCurrency"
            name="currentCurrency"
            onChange={[Function]}
            pattern="/^-?[0-9]+(\\\\.[0-9][0-9]?)?$/g"
            placeholder="0.00"
            type="number"
          />
        </label>
      </form>
      <div
        className="layout"
      >
        <p
          className="text"
        >
          You have 
          <span
            data-testid="account-balance"
          >
            £
            58.33
          </span>
        </p>
      </div>
    </section>
  </section>
</div>
`;
